### PR TITLE
feat:[CDS-57681]: GCS and Bamboo Artifact Source Support for TAS Swimlane NG

### PR DESCRIPTION
**Pull request by piyush-bhuwalka**
## Describe your changes

GCS and Bamboo Artifact Source Support for TAS Swimlane. UI changes will be merged later

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [x] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger utAll`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/45504)

### CODE REVIEW

**Review by shivamnegi94 — APPROVED** [review]

**Review by shivamnegi94 — COMMENTED** [review]
Please Add UT

**Comment by piyush-bhuwalka** [comment]
trigger feature-build

**Comment by piyush-bhuwalka** [comment]
trigger ut6

**Comment by piyush-bhuwalka** [comment]
trigger feature-build

File: 960-api-services/src/main/java/io/harness/googlecloudstorage/GcsHelperService.java [hunk 1]
```diff
@@ -152,4 +155,23 @@ private BuildDetails mapBlobToBuildDetails(Blob blob) {
...
+
+    try {
+      Blob blob = storage.get(BlobId.of(gcsInternalConfig.getBucket(), fileName));
+      InputStream inputStream = new ByteArrayInputStream(blob.getContent());
+      return inputStream;
+
+    } catch (Exception e) {
+      log.error("Exception occurred while fetching file {}", fileName, e);
+      return null;
+    }
```
Comment (PragyeshMishra01) [inline 960-api-services/src/main/java/io/harness/googlecloudstorage/GcsHelperService.java:175]: do we need to fail incase of exception?

**Review by PragyeshMishra01 — APPROVED** [review]

**Review by RishabhGupta34 — APPROVED** [review]
> Reviewed 1 of 16 files at r1, 1 of 5 files at r2, 11 of 13 files at r3, 3 of 4 files at r4, 1 of 1 files at r5, 1 of 1 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/45504)* status: 18 file(s) to review. 1. unresolved discussions. 1 of 2 approvals obtained.

**Comment by piyush-bhuwalka** [comment]
trigger feature-build

**Review by PragyeshMishra01 — APPROVED** [review]

**Review by sourabh-awashti — APPROVED** [review]
> Reviewed 1 of 16 files at r1, 1 of 5 files at r2, 11 of 13 files at r3, 3 of 4 files at r4, 1 of 1 files at r5, 1 of 1 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/45504)* status: 1. unresolved discussions.

**Comment by piyush-bhuwalka** [comment]
*[`960-api-services/src/main/java/io/harness/googlecloudstorage/GcsHelperService.java` line 175 at r3](https://reviewable.io/reviews/harness/harness-core/45504#-NRBwobvU6mrmlf7Sy-g-r3-175:-NRrvS_7EUBdseQ6eo2D:b-yof4x2) ([raw file](https://github.com/harness/harness-core/blob/a758669d80d4e258d98459b31ecedd5981c647b1/960-api-services/src/main/java/io/harness/googlecloudstorage/GcsHelperService.java#L175)):*
<details><summary><i>Previously, piyush-bhuwalka wrote…</i></summary><blockquote>

Yes now we throw exception
</blockquote></details>

Now throwing exception

**Review by PragyeshMishra01 — APPROVED** [review]
> Reviewed 1 of 16 files at r1, 1 of 5 files at r2, 11 of 13 files at r3, 3 of 4 files at r4, 1 of 1 files at r5, 1 of 1 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/45504)* status: :shipit: complete! ✓

**Pull request merged**